pymc4/distributions/distribution.py

```diff
@@ -27,7 +28,16 @@ class Distribution(Model):
     """Statistical distribution."""
 
     def __init__(
-        self, name: Optional[NameType], *, transform=None, observed=None, plate=None, **kwargs
+        self,
+        name: Optional[NameType],
+        *,
+        transform=None,
+        observed=None,
+        plate=None,
```

Comment:
One thing we should definitely clarify in the doc, is how `plate`/`plate_events` are the high level user API, while `sample_shape` and `plate_shape` is the internal (and lower level TFP) API depending on the resample/reshape distribution.

Reply:
Should we still call it plate I wonder and not side with TFP's naming convention?

Reply:
I think that we can't call it either `batch_shape` or `event_shape` because, with either name, it would give the impression to the user that it is setting the full batch or event shape of the distribution, instead of just appending some dimensions to either of them. I think that it needs to have a different name, and plate seems nice.

Reply:
We could maybe call it something like `stack_batch` or `stack_event` because that gives the impression that you are stacking things to either the batch or event dimensions

Reply:
DAG jargon like `plate` should be avoided, I think. It should be clear to the user what it represents. Something like `dist_sample_shape`? This links it to the distribution, and describes it more unambiguously.

Reply:
would `indep_shape` and `sample_shape` make sense?

Reply:
@lucianopaz thoughts?

Reply:
Sorry that I missed the notifications of the previous comments. @fonnesbeck, so we should change the names of the `plate` stuff.
@twiecki, I don't really like `sample_shape` because it will be confusing for the predictive sampling methods. At least I find that the `tfd.Sample` naming is very confusing and I wouldn't like to copy it in pymc4.
@junpenglao, I'm at a loss for an appropriate name for these. The correct term for the operation that we are carrying out could be either stacking or repeating. With `plate` and `plate_events` we either stack or repeat events or batches. I would call it `event_stack` and `batch_stack`, and we could then change the `Plate` distribution to `BatchStacker`. What do you guys think?

Reply:
I like `event_stack` and `batch_stack` and `BatchStacker`. Let's go with that if no one is opposed, we will probably not find the perfect name anyway.

Reply:
+1, I would be ok with plate as well actually. As long as we document it clearly even if we make a mistake user would adapt (hopefully).

Reply:
Great, I'll change the names and fix the conflicts tonight so we can merge this